swap_meet/vendor.py

```diff
@@ -1,2 +1,95 @@
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        self.inventory = inventory if inventory is not None else []
+
+    def add(self,item):
+        inventory = self.inventory.append(item)
+        return item
+
+    def remove(self,item):
+        if item in self.inventory:
+            inventory = self.inventory.remove(item)
+            return item
+        return False
+    
+    def get_by_id(self,id):
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None         # Returns None if no match is found
+    
+    
+    def swap_items(self, other_vendor, my_item, their_item): # test fatimah.swap_items(jolie, item_b, item_d) python automatically provide self.
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+    
+        self.inventory.remove(my_item)
+        other_vendor.inventory.remove(their_item)
+    
+        self.inventory.append(their_item)
+        other_vendor.inventory.append(my_item)
```

Comment:
Not at all required, but rather a neat little trick: The remove method you created will return the item, which we could then append into our list all at once if we wanted to!

```suggestion 
        self.inventory.append(other_vendor.inventory.remove(their_item))
        other_vendor.inventory.append(self.inventory.remove(my_item))
```